### PR TITLE
Problem: Windows build broken because of multiple issues

### PR DESCRIPTION
**Pull request by kurdybacha**
Windows issues:
* missing includes files
  Solution: added missing <memory> and <unordered_map>
  Here <map> was replaced with <unordered_map> as there is no need for
  sorted map.
* googletest fails because deprecation warning causing errors.
  Solution: D_SILENCE_TR1_NAMESPACE_DEPRECATION_WARNING option added.
* googletest fails because by default, new Visual Studio projects
  link the C runtimes dynamically but Google Test links them statically.
  Solution: gtest_force_shared_crt=ON option added.

Two last point should fix issue #199.

Besides adding appveyor.yml configuration to add Windows build to
CI in order to prevent accidental Windows build breakage on merge.
For now only Debug configuration as Release requires more time to figure
out.

### CODE REVIEW

**Comment by bluca** [comment]
I've enabled appveyor let's see if it runs

**Comment by bluca** [comment]
Building libzmq on appveyor is extremely slow for some reasons...
If possible, you might want to disable building the tests as that might speed it up - as opposed to autotools, it looks like with cmake the tests are always built with "make" rather than "make test/check"

**Comment by bluca** [comment]
The test fails:

```
1: Test command: C:\projects\cppzmq\Build\bin\Debug\unit_tests.exe
1: Test timeout computed to be: 10000000
1/1 Test #1: unit .............................***Exception: Exit code 0xc0000135
  0.20 sec
```

**Comment by kurdybacha** [comment]
@bluca build it is really slow indeed, don't know why though yet.
I will check in the evening and come back on this. Thanks.

**Comment by bluca** [comment]
Note that the slow build is a problem in libzmq's repository as well, so it's nothing to do with the appveyor file you added. Try with ```-DBUILD_TESTS=OFF```

**Comment by kurdybacha** [comment]
I disabled tests and perfs so build time is 2x faster now (~5min). Proper way would be to "cache" googletest and libzmq build between runs. Until then I am thinking to disable VS 2015.
Failing ctest is still pending investigation.

**Comment by bluca** [comment]
If you could implement ccache it would be fantastic, please also port it to libzmq and zproject :-)
It takes more than an hour right now to get a full run of the libzmq appveyor tests...

**Comment by kurdybacha** [comment]
@bluca Windows build now passes on my branch. Should be fine here in a bit...

File: zmq.hpp [hunk 1]
```diff
@@ -1107,7 +1108,7 @@ namespace zmq
     private:
         void *poller_ptr;
         std::vector<zmq_poller_event_t> poller_events;
-        std::map<socket_t*, handler_t> handlers;
+        std::unordered_map<socket_t*, handler_t> handlers;
```
Comment (sigiesec) [inline zmq.hpp]: I don't see how this change is related to the Windows build support (maybe #include <map> was missing?).

**Pull request merged**